lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -1077,6 +1085,51 @@ def build_accessible_object_scope(associations, options = {})
       end
     end
 
+    # Performance optimized function for PostgreSQL
+    def accessible_objects_function(user_id, operation, options)
+      accessible_objects_for_operations_function(user_id, [operation], options).values.flatten
+    end
+
+    # Performance optimized function for PostgreSQL
+    def accessible_objects_for_operations_function(user_id, operations, options)
+      prohibition_names = []
+
+      operation_names = operations.map do |o|
+        name = operation_identifier(o)
+        prohibition_names << prohibition_identifier(name) unless options[:ignore_prohibitions]
+        name
+      end
```

Comment:
the old code was using this `prohibitions_for` func to get the prohib names which did a `.where` to grab only those that exist in the DB.. since we are not even using this functionality (we always use `ignore_prohibitions=true`) perhaps we shouldn't change that behavior?  it's also kinda icky to build some second array in the `.map`ping of some first array.

Reply:
i would also be ok with raising a `NotImplementedError` if `ignore_prohibitions!=true`